src/main/webapp/WEB-INF/jsp/managestudy/viewStudySubject.jsp

```diff
@@ -1767,6 +1787,19 @@
             jQuery.blockUI({message: jQuery('#editSubjectForm'), css: {left: "300px", top: "10px"}});
         });
 
+        jQuery('#btn-copy button').click(function() {
+          var accessCode = document.getElementById('access-code-input');
+          accessCode.select();
+          accessCode.setSelectionRange(0, 99999); // for mobile devices
+          var copied = document.execCommand("copy");
+          if (copied) {
+            jQuery('#copy-success').show();          
+          } else {
+            alert('<fmt:message key="copy_to_clipboard_failed" bundle="${resword}"/>');            
```

Comment:
Even for failed copy, we should show the message below the input box instead of an alert

Reply:
@svadla-oc OK revised in 74db930